packages/atomic/src/components/commerce/atomic-commerce-result-list/atomic-commerce-result-list.tsx

```diff
@@ -0,0 +1,137 @@
+import {
+  buildProductListing,
+  ProductListingState,
+  ProductListing,
+  buildSearch,
+  SearchState,
+  Search,
+  Product,
+} from '@coveo/headless/commerce';
+import {Component, Element, Prop, State, h} from '@stencil/core';
+import StarIcon from '../../../images/star.svg';
+import {
+  BindStateToController,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../utils/initialization-utils';
+import {CommerceBindings} from '../atomic-commerce-interface/atomic-commerce-interface';
+
+/**
+ * The `atomic-commerce-result-list` component is responsible for displaying product results.
+ */
+@Component({
+  tag: 'atomic-commerce-result-list',
+  styleUrl: 'atomic-commerce-result-list.pcss',
+  shadow: true,
+})
+export class AtomicCommerceResultList
+  implements InitializableComponent<CommerceBindings>
+{
+  @InitializeBindings() public bindings!: CommerceBindings;
+  public productListing!: ProductListing;
+  public search!: Search;
+
+  @Element() public host!: HTMLDivElement;
+
+  @BindStateToController('productListing')
+  @State()
+  private productListingState!: ProductListingState;
+  @BindStateToController('search')
+  @State()
+  private searchState!: SearchState;
+  @State() public error!: Error;
+
+  /**
+   * The desired layout to use when displaying results. Layouts affect how many results to display per row and how visually distinct they are from each other.
+   */
+  @Prop({reflect: true}) display: 'grid' | 'list' = 'grid';
+
+  /**
+   * The spacing of various elements in the result list, including the gap between results, the gap between parts of a result, and the font sizes of different parts in a result.
+   */
+  @Prop({reflect: true}) density: 'normal' | 'compact' = 'normal';
+
+  /**
+   * The expected size of the image displayed in the results.
+   */
+  @Prop({reflect: true}) imageSize: number = 400;
+
+  public initialize() {
+    if (this.host.innerHTML.includes('<atomic-result-children')) {
+      console.warn(
+        'Folded results will not render any children for the "atomic-result-list". Please use "atomic-folded-result-list" instead.'
+      );
+    }
+
+    if (this.bindings.interfaceElement.type === 'product-listing') {
+      this.productListing = buildProductListing(this.bindings.engine);
+      this.productListing.refresh();
+    } else if (this.bindings.interfaceElement.type === 'search') {
+      this.search = buildSearch(this.bindings.engine);
+    }
+  }
+
+  private generateStarRating(rating: number) {
+    const stars = [];
+    for (let i = 0; i < 5; i++) {
+      stars.push(
+        <atomic-icon
+          class="w-5"
+          style={
+            i < rating ? {fill: 'gray'} : {fill: 'white', stroke: 'lightgray'}
+          }
+          icon={StarIcon}
+        ></atomic-icon>
+      );
+    }
+
+    return stars;
+  }
+
+  public render() {
+    const products =
```

Comment:
There is a bunch of functional components I've added to handle these "correctly". If you look at `atomic-result-list`, you'll see things like `ItemListGuard`, `DisplayWrapper`, `Placeholders` etc.